### ISSUE TITLE
Multiple async configuration approach is not working

### ISSUE THREAD

**Issue by maksimkurb** [nestjsx/nest-bull#161]
I have tried [your approach](https://github.com/nestjsx/nest-bull/issues/13#issuecomment-480098920) to define multiple queues but got an exception:
```
[Node] [Nest] 18812   - 2019-08-16 8:26 PM   [ExceptionHandler] Nest can't resolve dependencies of the BullQueueOptions_requests (?). Please make sure that the argument at index [0] is available in the BullModule context. +2ms
[Node] Error: Nest can't resolve dependencies of the BullQueueOptions_requests (?). Please make sure that the argument at index [0] is available in the BullModule context.
[Node]     at Injector.lookupComponentInExports (D:\Dev\dq\backend\node_modules\@nestjs\core\injector\injector.js:183:19)
[Node]     at process._tickCallback (internal/process/next_tick.js:68:7)
[Node]     at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
[Node]     at startup (internal/bootstrap/node.js:283:19)
[Node]     at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

```ts
function createBullModule(queueName: string) {
  return {
    name: queueName,
    // FIXME: https://github.com/fwoelffel/nest-bull/issues/88
    useFactory: (config: ConfigService) => ({
      name: queueName,
      options: {
        redis: {
          host: config.redisHost,
          port: config.redisPort,
          password: config.redisPassword,
        },
      },
    }),
    imports: [ConfigModule],
    inject: [ConfigService],
  }
}

const requestsQueue = createBullModule('requests')
const billingPeriodBonusQueue = createBullModule('billing')

const BullQueueModule = BullModule.forRootAsync([
  requestsQueue,
  billingPeriodBonusQueue,
])

@Module({
  imports: [BullQueueModule],
  exports: [BullQueueModule],
})
export class QueueModule {}
```

**Comment by maksimkurb** [comment]
In my opinion this is caused because of [this code snippet](https://github.com/nestjsx/nest-bull/blob/08f1fe15bdcbb47b73718148e97d2de2be97767e/lib/bull.module.ts#L35-L40):
```ts
    const imports =
      optionsArr
        .map(option => option.imports)
        .reduce((acc = [], i) => {
          acc.push(i);
        }) || [];
```

`reduce` function always returns `undefined`, so it always will return an empty array `[]`.

After I've changed this lines to:
```ts
        const imports = optionsArr
            .map(option => option.imports)
            .reduce((acc = [], i) => {
               return acc.concat(i); // 1 - return, 2 - replace push via concat or we cat get array of arrays [[imports1], [imports2]]
            })
            .filter((v, i, a) => a.indexOf(v) === i) || []; // filter out duplicates, don't sure if it is needed
```

all started working correctly.

**Comment by maksimkurb** [comment]
Made a [PR](https://github.com/nestjsx/nest-bull/pull/162) for it

**Comment by smonv** [comment]
I got the same problem, whenever I add a second forRootAsync config, my first queue start couldn't resolve dependencies.

@fwoelffel Can you review #162  and release a fix soon ?

**Comment by fwoelffel** [comment]
Hi @maksimkurb @smonv 
Thanks for your help :slightly_smiling_face: I've just released v0.8.3 which fixes this

**Comment by smonv** [comment]
@maksimkurb Thank for the fix
@fwoelffel Thank for release it.
